optimade/validator/validator.py

```diff
@@ -261,17 +261,23 @@ def validate_implementation(self):
         self._log.debug("Testing base info endpoint of %s", info_endp)
 
         # Get and validate base info to find endpoints
+        # If this is not possible, the exit at this stage
         base_info = self._test_info_or_links_endpoint(info_endp)
         if not base_info:
-            raise RuntimeError(
-                "Unable to deserialize base info (see previous errors), so cannot continue."
+            self._log.critical(
+                f"Unable to deserialize response from introspective {info_endp!r} endpoint. "
+                "This is required for all further validation, so the validator will now exit."
             )
+            self.print_summary()
+            return
 
-        self.provider_prefix = (
-            base_info.dict()["meta"].get("provider", {}).get("prefix")
-        )
+        # Grab the provider prefix from base info and use it when looking for provider fields
+        self.provider_prefix = None
+        meta = base_info.get("meta", {})
+        if meta.get("provider") is not None:
+            self.provider_prefix = meta["provider"].get("prefix")
```

Comment:
Why is this not fine? This should still return `None` if none of this can be found.

I guess the real question is, in which case will `provider_prefix` not be `None` or the value of `meta.provider.prefix`?

```suggestion
        self.provider_prefix = base_info.get("meta", {}).get("provider", {}).get("prefix")
```

Reply:
Annoyingly, because `provider` is an optional field in `meta`, if it is missing then it gets set to `None` by pydantic (c.f. all the `exclude_unset` stuff we've discussed in the past), so the `.get("provider", {})` will still return `None` and fallover.

Reply:
You could go through `meta.provider.prefix` of the serialized model, as I think you are alluding to, but that reduces the flexibility of other parts of the validator (i.e. running as many tests as possible even if deserialization fails in a minor way)

Reply:
Nahh, that's fine. Didn't even mean this, I just wanted to quickly write out the location of `prefix`.

Reply:
Although we might think about utilizing the pydantic models more... But I think we have discussed this many times already..